pkg/webui/components/form/index.js

```diff
@@ -23,6 +24,30 @@ import FormField from './field'
 import FormSubmit from './submit'
 
 class InnerForm extends React.PureComponent {
+  constructor (props) {
+    super(props)
+    this.focusGroup = React.createRef()
+  }
+
+  componentDidUpdate (prevProps) {
+    const { formError, isSubmitting, isValid } = this.props
+    const { isSubmitting: prevIsSubmitting, formError: prevFormError } = prevProps
+
+    // Scroll form notification into view if needed
+    if (formError && !prevFormError) {
+      scrollIntoView(this.focusGroup.current, { behavior: 'smooth' })
```

Comment:
can we focus on the first field as well?

Reply:
I think that can lead to the false assumption that the first field is relevant for resolving the error. Also, for error's like "An unknown error occurred. Please try again later.", there is no remedy in changing the form values.

What we can do though, is laying focus in the notification container, so that another tab will put the user on the first input of the form again.

Reply:
> I think that can lead to the false assumption that the first field is relevant for resolving the error.

This is not about making assumptions about which field causes the backend to respond with an error. Scrolling to the top of the form without moving focus to the visible area is poor accessibility. For example, I always use `Tab`, `Tab+Shift` to navigate through forms. This implementation causes the viewport of my browser to jump back to the submit section and forces me to go through all the fields back to the top.

> What we can do though, is laying focus in the notification container

How do you plan to focus on `div`? Using `tabIndex` ?

Reply:
I agree with you on moving the focus. I just don't agree with putting the focus on the input element, because IMO it would suggest to the user that a different input needs to be made, which we don't know. That's why I suggested putting it on the notification container.

> How do you plan to focus on `div`? Using `tabIndex`?

Yes. Putting a negative `tabIndex` on an element [will make it focusable via JS](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex).

See my implementation in https://github.com/TheThingsNetwork/lorawan-stack/pull/931/commits/bc8d9c460f55710240dd244cece02be98939c878 .

Reply:
i agree. do you want to leave the default outline?
<img width="635" alt="Screenshot 2019-07-16 at 15 09 44" src="https://user-images.githubusercontent.com/16374166/61306999-ca062980-a7ed-11e9-93e9-7b7d8d225add.png">

Reply:
Ah, didn't experience that. Will make sure there will be no outline.

Reply:
Removed the outline.